finder-app/writer.c

```diff
@@ -0,0 +1,84 @@
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <syslog.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <libgen.h>
+#include <errno.h>
+
+int main(int argc, char* argv[])
+{
+	openlog(NULL, 0, LOG_USER);
+
+	/* Exit if number of arguments do not match writer requirements */
+	if(argc != 3)
+	{
+		syslog(LOG_ERR, "Error: Please provide two arguments for the path-to-file and string to write\n");
+		syslog(LOG_ERR, "Usage: ./writer [path-to-file] [string-to-write]\n");	
+		exit(1);
+	}
+
+	char *path 	= argv[1];
+	char *writestr	= argv[2];
+	char openpath[200];
```

Comment:
@DhruvHMehta how do you know 200 bytes is enough?

Reply:
Thanks for the feedback, Dan. I should have used malloc or the PATH_MAX instead. I will make these changes before using the writer for Assignment 3.